client/src/test/java/org/asynchttpclient/ClientStatsTest.java

```diff
@@ -0,0 +1,169 @@
+package org.asynchttpclient;
+
+import static org.asynchttpclient.Dsl.asyncHttpClient;
+import static org.asynchttpclient.Dsl.config;
+import static org.testng.Assert.assertEquals;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.testng.annotations.Test;
+
+/**
+ * Created by grenville on 9/25/16.
+ */
+public class ClientStatsTest extends AbstractBasicTest {
+
+    @Test(groups = "standalone")
+    public void testClientStatus() throws Throwable {
+        try (final DefaultAsyncHttpClient client = (DefaultAsyncHttpClient) asyncHttpClient(config().setKeepAlive(true).setPooledConnectionIdleTimeout(5000))) {
+            final String url = getTargetUrl();
+
+            final ClientStats emptyStats = client.getClientStats();
+
+            assertEquals(emptyStats.toString(), "There are 0 total connections, 0 are active and 0 are idle.");
+            assertEquals(emptyStats.getActiveConnectionCount(), 0);
+            assertEquals(emptyStats.getIdleConnectionCount(), 0);
+            assertEquals(emptyStats.getTotalConnectionCount(), 0);
+
+            final List<ListenableFuture<Response>> futures = new ArrayList<>();
```

Comment:
I'll be honest: I rarely use Java 8 streams as I mostly use Scala, but isn't there a way to have this more concise with Java 8?